runtimes/common/src/dip/did.rs

```diff
@@ -0,0 +1,100 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2023 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use did::did_details::DidDetails;
+use frame_system::pallet_prelude::BlockNumberFor;
+use kilt_dip_support::merkle::RevealedWeb3Name;
+use pallet_did_lookup::linkable_account::LinkableAccountId;
+use pallet_dip_provider::traits::IdentityProvider;
+use parity_scale_codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use sp_std::vec::Vec;
+
+#[derive(Encode, Decode, TypeInfo)]
+pub enum LinkedDidInfoProviderError {
+	DidNotFound,
+	DidDeleted,
+	Internal,
+}
+
+impl From<LinkedDidInfoProviderError> for u16 {
```

Comment:
The error can be maximum a `u16`. If nested errors are thrown, they should not overlap with each other, that's why I went for a `u8` for nested errors. Not sure what you refer by scale encoding.

Reply:
So you need that for the Error in the dip-provider pallet?

And this doesn't work since the marco doesn't support to bind `T: Config`
```
	#[pallet::error]
	pub enum Error<T> {
		/// The specified commitment cannot be found.
		CommitmentNotFound,
		/// Error when retrieving the identity details of the provided subject.
		IdentityProvider(u16),
		/// Error when generating a commitment for the retrieved identity.
		IdentityCommitmentGenerator(
			<<T as Config>::IdentityCommitmentGenerator as IdentityCommitmentGenerator<T>>::Error,
		),
		/// Error inside the external hook logic.
		Hook(u16),
	}
```

Reply:
Yes, this does not work, unfortunately. There is a static assertion that the whole `RuntimeError` must be 4-bytes long at most. Which leaves 3 bytes for pallet errors, and 2 bytes for sub-errors.

Reply:
yeah we had the same issue in the DID pallet if i remember correctly. :/